lib/decompress/zstd_decompress_block.c

```diff
@@ -2125,6 +2125,9 @@ ZSTD_decompressBlock_internal(ZSTD_DCtx* dctx,
         srcSize -= seqHSize;
 
         RETURN_ERROR_IF(dst == NULL && nbSeq > 0, dstSize_tooSmall, "NULL not handled");
+        RETURN_ERROR_IF(dstCapacity == 0 && nbSeq > 0, dstSize_tooSmall, "dstCapacity == 0, but nbSeq is non-zero");
```

Comment:
nit: We can combine this check with the line `2127`. `dst` may only be `NULL` if `dstCapacity == 0`.

Reply:
Would it be possible for `dst == NULL` and `dstCapacity == 0` but `nbSeq > 0`? And would this be caught elsewhere?